bin/api_diff.ml

```diff
@@ -1,55 +1,42 @@
 let tool_name = "api-diff"
 
+type mode = Unwrapped | Wrapped of string | Cmi
+
+let rec mode ~reference ~current ~main_module ~unwrapped =
+  match (both_directories reference current, main_module, unwrapped) with
+  | Ok true, Some main_module, false -> Ok (Wrapped main_module)
+  | Ok true, None, true -> Ok Unwrapped
+  | Ok false, None, false -> Ok Cmi
+  | _ ->
```

Comment:
It could have been nice to split this case in several in order to give more specific message. But at the same time, a single message with all alternatives is fine!